src/app/clusters/mode-base-server/mode-base-server.cpp

```diff
@@ -0,0 +1,445 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <app-common/zap-generated/attributes/Accessors.h>
+#include <app/InteractionModelEngine.h>
+#include <app/clusters/mode-base-server/mode-base-server.h>
+#include <app/clusters/on-off-server/on-off-server.h>
+#include <app/reporting/reporting.h>
+#include <app/util/attribute-storage.h>
+#include <platform/DiagnosticDataProvider.h>
+
+// using namespace std;
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using chip::Protocols::InteractionModel::Status;
+using BootReasonType = GeneralDiagnostics::BootReasonEnum;
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace ModeBase {
+
+std::map<uint32_t, Instance *> Instance::ModeBaseAliasesInstanceMap;
+
+bool Instance::HasFeature(Feature feature) const
+{
+    return (mFeature & to_underlying(feature)) != 0;
+}
+
+bool Instance::isAliasCluster() const
+{
+    for (unsigned int AliasedCluster : AliasedClusters)
+    {
+        if (mClusterId == AliasedCluster)
+        {
+            return true;
+        }
+    }
+    return false;
+}
+
+CHIP_ERROR Instance::Init()
+{
+    // Check that the cluster ID given is a valid mode base alias cluster ID.
```

Comment:
It should be the caller's problem. The impl **should not** have to know about all possible aliases.

Reply:
But all the possible aliases are hard set in the spec, so we can hard code them in. It would just be another nice check for the SDK user more than anything.